envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/containers/LuaTestsContainer.kt

```diff
@@ -0,0 +1,52 @@
+package pl.allegro.tech.servicemesh.envoycontrol.config.containers
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import org.testcontainers.containers.BindMode
+import org.testcontainers.containers.GenericContainer as BaseGenericContainer
+
+class LuaTestsContainer : BaseGenericContainer<LuaTestsContainer>("vpanov/lua-busted@$hash") {
+
+    companion object {
+        const val hash = "sha256:21676428f30907d4081b5c02cae12e952b6de5bef544643c9eeaf2b416eccb70"
+        private val mapper = ObjectMapper()
+            .registerModule(KotlinModule())
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    }
+
+    private val LUA_SRC_DIR = "lua"
+    private val LUA_SRC_DIR_DEST = "/lua"
+    private val LUA_SPEC_DIR = "lua_spec"
+    private val LUA_SPEC_DIR_DST = "/lua_spec"
+
+    override fun configure() {
+        super.configure()
+        withClasspathResourceMapping(LUA_SRC_DIR, LUA_SRC_DIR_DEST, BindMode.READ_ONLY)
+        withClasspathResourceMapping(LUA_SPEC_DIR, LUA_SPEC_DIR_DST, BindMode.READ_ONLY)
+        withWorkingDirectory(LUA_SPEC_DIR_DST)
+        withCommand("sh", "-c", "sleep 180")
```

Comment:
why do we sleep 180?

Reply:
It is an ugly workaround to run "one shot" container in testcontainers. Just look a few lines below, you will see we run actual command using `execInContainer("busted", "--output=json", "--lpath=$LUA_SRC_DIR_DEST/?.lua", LUA_SPEC_DIR_DST)`.

one shot container = Container we launch to only run one command and exit (and get output).

If you know any better way to do it, let me know.

Reply:
Have you tried putting everything from `execInContainer` into `withCommand`? CMD / ENTRYPOINT becomes pid 1 and when the process exits the container will as well.

Reply:
I played around with it and there is no easy way to get failing tests (setting `withCommand` + https://www.testcontainers.org/features/startup_and_waits/#one-shot-startup-strategy-example will make container not start in case of a failed test and we do not know which test failed).

Reply:
the main problem with putting everything in `withCommand` is no easy way to get an output.  `execInContainer` provide an easy access to command output